src/modules/sub-items/components/table-view/table.view.component.js

```diff
@@ -18,140 +20,42 @@ export default class TableViewComponent extends Component {
         this.sortByDate = this.sortByDate.bind(this);
         this.sortByPriority = this.sortByPriority.bind(this);
         this.renderItem = this.renderItem.bind(this);
-        this.onSelectAll = this.onSelectAll.bind(this);
-
-        this.state = {
-            items: props.items,
-            isAscSort: false,
-            sortKey: false,
-        };
-    }
-
-    UNSAFE_componentWillReceiveProps({ items }) {
-        this.sortItems(items);
+        this.selectAll = this.selectAll.bind(this);
     }
 
     /**
-     * Sorts provided items
-     *
-     * @method sortItems
-     * @param {Array} items
-     * @memberof TableViewComponent
+     * Changes sort to name
      */
-    sortItems(items) {
-        const sortKey = this.state.sortKey;
-
-        if (!sortKey) {
-            this.setState(() => ({ items }));
-        }
+    sortByName() {
+        const { onSortChange } = this.props;
 
-        if (sortKey === SORTKEY_NAME) {
-            this.sortByName(items);
-        } else if (sortKey === SORTKEY_DATE) {
-            this.sortByDate(items);
-        } else if (sortKey === SORTKEY_PRIORITY) {
-            this.sortByPriority(items);
-        } else {
-            this.setState(() => ({ items }));
-        }
+        onSortChange('ContentName');
```

Comment:
I would make it a constant like `const KEY_CONTENT_NAME = 'ContentName';`. Then in the:

```
const SORTKEY_MAP = {
    [KEY_CONTENT_NAME]: 'name',
    DateModified: 'date',
    LocationPriority: 'priority',
};
```

Reply:
The similar rule applies to other places.